xapian-letor/api/feature_internal.cc

```diff
@@ -0,0 +1,181 @@
+/** @file feature_internal.cc
+ * @brief Definition of Feature::Internal class.
+ */
+/* Copyright (C) 2019 Vaibhav Kansagara
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include "xapian-letor/feature.h"
+#include "feature_internal.h"
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include "debuglog.h"
+
+using namespace std;
+using namespace Xapian;
+
+Database
+Feature::Internal::get_database() const
+{
+    LOGCALL(API, Database, "Feature::Internal::get_database", NO_ARGS);
+    return feature_db;
```

Comment:
I would inline all these trivial getters and setters in the header - that way they're exposed to the compiler to optimise better.  We won't have logging for them, but I've found logging is of limited interest in such methods anyway.

Reply:
fixed in a fixup commit.